### ISSUE TITLE
generalizing "enough" constructions

### ISSUE THREAD

**Issue by cbonial** [amrisi/amr-guidelines#175]
A couple of weeks ago the little Prince sentence "...I had scarcely enough drinking water to last a week." came up in our discussion, which focused on "last" in this context.  I had asked how "enough" would generally be annotated in cases like this, since this type of "enough" construction can occur with a variety of verbs, not just "last": "I drank enough to satisfy my thirst."  These are quite similar to some of the degree, and degree-consequence constructions that we had looked at in the past in relation to have-degree-91.  Here, however, instead of dealing a gradable property or state of something (generally captured by an adjective with have-degree-91), we're focused on a gradable quantity.  In the past, I had proposed that these could also be covered by have-degree-91, at least where a comparison was evoked; for example: 

Have-degree-91
Arg1: attribute (e.g. tall)
Arg2: domain, entity characterized by attribute (e.g. girl)
Arg3: degree itself (e.g. more, less, equal)
Arg4: compared-to (e.g. (than the) BOY)
Arg5: consequence, result of degree (e.g. (not tall enough) TO RIDE THE ROLLERCOASTER)

![as much beer](https://cloud.githubusercontent.com/assets/4305080/13406966/24349278-def4-11e5-9e18-d39c4485c26e.PNG)

This is essentially treating the quantity itself as the gradable property, while the thing with the quantity-property is the domain, arg2.  To me, this is ok, but certainly not abundantly intuitive.  Perhaps we should consider adding roles to have-quant-91 to parallel have-degree-91?  Like gradable properties, quantities are often compared, and also have consequences/results, as in "Enough X to Y," or "too much X to Y" etc.

If we want to stick with use of have-degree-91, then these "enough" constructions involving quantities could be annotated like this: 

![enough degree consequence](https://cloud.githubusercontent.com/assets/4305080/13407188/19bc77d8-def5-11e5-9fc9-f1a5c5e257bd.PNG)

As you can see, there are some redundancies to this as well -- repetition of water variable. If, on the other hand, we introduced parallel compared-to and consequence args to the have-quant-91, then we could have this: 

![enough quant consequence](https://cloud.githubusercontent.com/assets/4305080/13407349/fc7e975e-def5-11e5-97c9-0b32e2472209.PNG)

I think this more accurately captures the fact that the quantity itself allows for another event/consequence/resulting state.  I don't think the current annotation that we looked at in our meeting captures this relationship between the quantity itself being enough/too little to satisfy a result. 

This would also allow us to go deeper into the semantics of related constructions using "too much, as much,"  etc. For example: 

![too much water](https://cloud.githubusercontent.com/assets/4305080/13408310/21e75e2c-defb-11e5-9bc5-15a67458441b.PNG)

To get a feel for how these are currently being annotated, consider this (somewhat weird) portion of an AMR sentence and its current annotation, as well as what could be represented with have-quant (or feasibly have-degree-91): 

![vision have-quant](https://cloud.githubusercontent.com/assets/4305080/13408325/395b5ad6-defb-11e5-9f43-95d2709d5134.PNG)

In summary, I think that these could be handled in the re-annotation done for have-degree-91 (many of which, by the way, can be found by simply looking for "enough to" or "too much too"), but perhaps it would be more intuitive and elegant to introduce parallel roles for have-quant-91.  Above all, it should be noted again that there are currently inconsistencies and problems in the existing annotations of these constructions, indicating that annotators simply don't know what to do with these infinitival phrases, which are arguments licensed by the constructions.
